### PR TITLE
Added posibility to change discussion-tree chevrons

### PR DESCRIPTION
**Pull request by c0r73x**

### CODE REVIEW

File: doc/gitlab.nvim.txt [hunk 1]
```diff
@@ -187,6 +187,7 @@ you call this function with no values the defaults will be used:
         temp_registers = {}, -- List of registers for backing up popup content (see `:h gitlab.nvim.temp-registers`)
       },
       discussion_tree = { -- The discussion tree that holds all comments
+        chevrons = { " ", " ", "  " },
```
Comment (jakubbortlik) [inline doc/gitlab.nvim.txt:190]: Hi @c0r73x, you could add some description to the newly added setting. It's not obvious what the third string is for (it's indentation, right?). Also, maybe "chevrons" may not be the most explicit name - imagine somebody decides to use something like `+` and `-` instead. E.g., [neo-tree](https://github.com/nvim-neo-tree/neo-tree.nvim) uses `expander_collapsed` and `expander_expanded`. You might consider changing this to something like (and add some more documentation):
```suggestion
        expanders = {
          expanded = " ", 
          collapsed = " ",
          indentation = "  ",
        },
```

Comment (c0r73x) [reply doc/gitlab.nvim.txt]: Done 👍

**Comment by harrisoncramer** [comment]
This works well, but currently but `after/syntax/gitlab.vim` file has a syntax match that includes the chevrons when looking for a username. Can we look into fixing that?

I'm not sure how this would be done, possibly by dynamically searching for the value set in the configuration? This is what it looks like right now.

```vim
syntax match Username "\%([]\)\@<= @\S*"
syntax match Mention "\%([] \)\@<!@\S*"
```

**Comment by harrisoncramer** [comment]
This is what I came up with. In `colors.lua` we can set global variables:

```lua
-- Set icons into global vim variables for syntax matching
local expanders = state.settings.discussion_tree.expanders
vim.g.gitlab_discussion_tree_expander_open = expanders.expanded
vim.g.gitlab_discussion_tree_expander_closed = expanders.collapsed
vim.g.gitlab_discussion_tree_draft = ''
vim.g.gitlab_discussion_tree_resolved = '✓'
vim.g.gitlab_discussion_tree_unresolved = '-'
```

And then in `syntax.vim` we can use those variables to match:

```vim
if filereadable($VIMRUNTIME . '/syntax/markdown.vim')
  source $VIMRUNTIME/syntax/markdown.vim
endif

syntax match Date "\v\d+\s+\w+\s+ago"
highlight link Date GitlabDate

execute 'syntax match Unresolved /\s' . g:gitlab_discussion_tree_unresolved . '\s\?/'
highlight link Unresolved GitlabUnresolved

execute 'syntax match Resolved /\s' . g:gitlab_discussion_tree_resolved . '\s\?/'
highlight link Resolved GitlabResolved

execute 'syntax match GitlabDiscussionOpen /^' . g:gitlab_discussion_tree_expander_open . '/'
highlight link GitlabDiscussionOpen GitlabChevron

execute 'syntax match GitlabDiscussionClosed /^' . g:gitlab_discussion_tree_expander_closed . '/'
highlight link GitlabDiscussionClosed GitlabChevron

execute 'syntax match Draft /' . g:gitlab_discussion_tree_draft . '/'
highlight link Draft GitlabDraft

execute 'syntax match Username "@\w\+"'
highlight link Username GitlabUsername

execute 'syntax match Mention "\%([' . g:gitlab_discussion_tree_expander_open . g:gitlab_discussion_tree_expander_closed . '] \)\@<! @\S\+"'
highlight link Mention GitlabMention

let b:current_syntax = "gitlab"
```

What do you guys think of this?

**Comment by c0r73x** [comment]
Looks good to me 👍

**Comment by c0r73x** [comment]
I added it to my branch, though i'm not sure about the syntax file. The old one had more syntax matches, maybe they werent used?

**Comment by harrisoncramer** [comment]
> I added it to my branch, though i'm not sure about the syntax file. The old one had more syntax matches, maybe they werent used?

Yeah I condensed them into one thing, and fixed up some of those changes directly on your branch! I'll merge this into develop now, it'll probably be in `main` in the next few days, I'll ship it along with the keymap changes.

Thanks for your contribution!

**Pull request merged**